src/fiatExchanges/SelectProvider.tsx

```diff
@@ -158,39 +166,88 @@ export default function SelectProviderScreen({ route, navigation }: Props) {
     asyncProviders.result?.externalProviders
   )
 
+  const supportOnPress = () => navigate(Screens.SupportContact)
+
+  const switchCurrencyOnPress = () =>
+    navigate(Screens.FiatExchangeCurrency, {
+      flow:
+        route.params.flow === CICOFlow.CashIn ? FiatExchangeFlow.CashIn : FiatExchangeFlow.CashOut,
+    })
+
   const exchanges = asyncExchanges.result ?? []
+  const legacyMobileMoneyProviders = asyncProviders.result?.legacyMobileMoneyProviders
+
+  const coinbasePayVisible =
+    coinbaseProvider &&
+    !coinbaseProvider.restricted &&
+    coinbasePayEnabled &&
+    appId &&
+    digitalAsset === CiCoCurrency.CELO
+
+  const anyProviders =
+    normalizedQuotes.length ||
+    coinbasePayVisible ||
+    exchanges.length ||
+    legacyMobileMoneyProviders?.length
 
   return (
-    <ScrollView>
-      <PaymentMethodSection
-        normalizedQuotes={normalizedQuotes}
-        paymentMethod={PaymentMethod.Card}
-        setNoPaymentMethods={setNoPaymentMethods}
-        flow={flow}
-      />
-      <PaymentMethodSection
-        normalizedQuotes={normalizedQuotes}
-        paymentMethod={PaymentMethod.Bank}
-        setNoPaymentMethods={setNoPaymentMethods}
-        flow={flow}
-      />
-      <LegacyMobileMoneySection
-        providers={asyncProviders.result?.legacyMobileMoneyProviders || []}
-        digitalAsset={digitalAsset}
-        flow={flow}
-      />
-      <CoinbasePaymentSection
-        digitalAsset={digitalAsset}
-        cryptoAmount={route.params.amount.crypto}
-        coinbaseProvider={coinbaseProvider}
-      />
-      <ExchangesSection
-        exchanges={exchanges}
-        selectedCurrency={route.params.selectedCrypto}
-        flow={flow}
-      />
-      <LimitedPaymentMethods visible={noPaymentMethods} flow={flow} />
-    </ScrollView>
+    <SafeAreaView style={styles.container}>
+      {anyProviders ? (
+        <ScrollView>
+          <PaymentMethodSection
+            normalizedQuotes={normalizedQuotes}
+            paymentMethod={PaymentMethod.Card}
+            setNoPaymentMethods={setNoPaymentMethods}
+            flow={flow}
+          />
+          <PaymentMethodSection
+            normalizedQuotes={normalizedQuotes}
+            paymentMethod={PaymentMethod.Bank}
+            setNoPaymentMethods={setNoPaymentMethods}
+            flow={flow}
+          />
+          <LegacyMobileMoneySection
+            providers={legacyMobileMoneyProviders || []}
+            digitalAsset={digitalAsset}
+            flow={flow}
+          />
+          <CoinbasePaymentSection
+            cryptoAmount={route.params.amount.crypto}
+            coinbaseProvider={coinbaseProvider}
+            visible={coinbasePayVisible}
+            appId={appId}
+          />
+          <ExchangesSection
+            exchanges={exchanges}
+            selectedCurrency={route.params.selectedCrypto}
+            flow={flow}
+          />
```

Comment:
what if there are no exchanges, but there are other kinds of providers? You got rid of the behavior where the message appears saying there aren't any options. Wasn't this section supposed to be hidden now if there were no exchanges?

Reply:
+1